charts/cloudserver-front/values.yaml

```diff
@@ -6,7 +6,7 @@ orbit:
   endpoint: https://api.zenko.io
 # When 'orbit.enabled' is 'true', these aren't used, please use
 # https://zenko.io to manage your deployment
-endpoint: zenko.local
+endpoint: zenko-cloudserver-front
```

Comment:
Why?

Reply:
Apparently something in the code is rejecting URLs with '.' (dots) in hostnames. @JianqinWang is it a Scality function or something else we can change because '.' (dots) in host name in pretty frequent

Reply:
The default settings weren't working, so we were changing these values to get a deployment running

Reply:
So, what's not clear to me: is `endpoint` a *hostname*, or is it a *hostname + dnsdomainname*? I believe it's likely to be the latter, so values with dots (like the default) *must* be accepted.

Strictly speaking, a *hostname* can't have a dot in it.

Reply:
`zenko-cloudserver-front` is the hostname of the cloudserver, but we can use hostname + dnsdomainname - we don't think the '.' is actually causing issues, but rather 'zenko.local' wasn't resolving to anything.

Reply:
Oh, right. Now what I don't get is why this value should be set to the server hostname, or should resolve at all (it's just used to dispatch vhosts, no?).
Also, hard-coding `zenko-cloudserver-front` wouldn't be right because nothing says that's the hostname for all deployments (it wouldn't be for mine, for example).

Reply:
Basically: can anyone explain what `ENDPOINT` *means* and how it's *used*?

Reply:
It shouldn't be hard coded - this branch was originally just to save progress/changes while testing deployments.
We set `endpoint` here: https://github.com/scality/Zenko/blob/ba6e26902ac23f71ca0c38adb41b74a3b4163e61/charts/cloudserver-front/templates/deployment.yaml#L47
which gets passed to here in S3: https://github.com/scality/S3/blob/5731a65ab400017f3be45639050d535e1d5ca177/docker-entrypoint.sh#L9
The code in the docker-entrypoint script changes the S3 `config.json`. In this case, it adds the hostname `zenko-cloudserver-front` to the restEndpoints.
I will remove this change so it's not hardcoded to `zenko-cloudserver-front`.

Reply:
Ah, thanks for the pointers/clarification/intel!

Looking at this, it seems that indeed, the `endpoint` is used to map a vhost (as used by the client, and set in the HTTP `Host` header) to an internal 'region'. As such, the value depends on how the service is exposed to the outside world (with DNS etc), not to the host where the service is running (though, granted, if you test from 'within' a K8s cluster, where DNS for the service object is provided, it may make sense to set it to the service object name).

I would however prefer to keep the `zenko.local` default, since we'll likely expose the service using an Ingress object, with a given host selector, and the ingress object configuration and the `endpoint` value must be kept in sync, as explained in the `README.md`.

Finally, given this, I'm even more confused as to why `endpoint` should be set when running an Orbit-managed system: in that case, Orbit should push the endpoint configuration into CloudServer, and no initial value should be required. If it is, that seems like a bug (or maybe design mistake).